extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -264,15 +264,29 @@ export class SchemaCompareResult {
 		return data;
 	}
 
+	private getFormattedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
+		// if there is no entry, the script has to be \n because an empty string shows up as a difference but \n doesn't
+		if ((getSourceScript && diffEntry.sourceScript === null)
+			|| (!getSourceScript && diffEntry.targetScript === null)) {
+			return '\n';
+		}
+
+		let script = this.getAggregatedScript(diffEntry, getSourceScript);
+		return script;
+	}
+
 	private getAggregatedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
 		let script = '';
 		if (diffEntry !== null) {
-			script += getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			let diffEntryScript = getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			if (diffEntryScript) {
```

Comment:
just for my understanding:

suppose a child script is just ' '  will this part all \n to that? because then when it goes to parent it might end up having multiple lines. am I missing something?

Reply:
Empty strings evaluate to false, so it doesn't go inside the if in that case. The scripts are trimmed in SqlToolsService so scripts that might be all whitespace would be empty strings at this point.